shiny/_utils.py

```diff
@@ -262,15 +262,15 @@ def private_seed() -> Generator[None, None, None]:
 
 
 def wrap_async(
-    fn: Callable[P, R] | Callable[P, Awaitable[R]],
+    fn: Callable[P, R] | Callable[P, Awaitable[R]] | Callable[P, Awaitable[R] | R],
```

Comment:
I think the change on this line shouldn't be necessary, after making the changes above in the signature of `__init__`.